rasterio/plot.py

```diff
@@ -105,31 +105,34 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
                 # will have the colors in the order expected by
                 # matplotlib (RGB)
                 arr = source.read(rgb_indexes, masked=True)
-                arr = reshape_as_image(arr)
-
             except KeyError:
-                arr = source.read(1, masked=True)
+                if indexes == None: indexes = (0,1,2)
```

Comment:
if you want a one-liner I would probably do `indexes = indexes or (0, 1, 2)`

Reply:
thanks for your nice suggestion, the scripts have been modified in the latest commit.